File: src/Animal.cpp
#include "Animal.h"

Animal::Animal(const std::string &name, const std::string &birthDate)
    : name(name), birthDate(birthDate) {}

void Animal::addCommand(const std::string &command) {
    commands.push_back(command);
}

std::vector<std::string> Animal::getCommands() const {
    return commands;
}

std::string Animal::getName() const {
    return name;
}

std::string Animal::getBirthDate() const {
    return birthDate;
}

File: 13-15/JAVA/src/animals/Animal.java
package animals;

import java.util.ArrayList;
import java.util.List;

public class Animal {
    private String name;
    private String birthDate;
    private List<String> commands;

    public Animal(String name, String birthDate) {
        this.name = name;
        this.birthDate = birthDate;
        this.commands = new ArrayList<>();
    }

    public void addCommand(String command) {
        commands.add(command);
    }

    public List<String> getCommands() {
        return new ArrayList<>(commands);
    }

    public String getName() {
        return name;
    }

    public String getBirthDate() {
        return birthDate;
    }
}
